snapcraft/plugins/catkin.py

```diff
@@ -68,6 +68,10 @@
       (string)
       The URI to ros master setting the env variable ROS_MASTER_URI. Defaults
       to http://localhost:11311.
+    - disable-parallel:
```

Comment:
This functionality actually already exists, just use [this](https://github.com/snapcore/snapcraft/blob/master/snapcraft/_baseplugin.py#L174) to enable it. The [make](https://github.com/snapcore/snapcraft/blob/master/snapcraft/plugins/make.py#L108) plugin is a great example.

Reply:
I see. I'll let CI finish the current run then open another PR to replace this.

Reply:
No need to close the PR, just revamp it a bit. Make sure you add a test that verifies that specifying the option actually does result in `-j1`.

Reply:
Well, the scope of the feature slightly change and most of the introduced code goes down the drain. I think opening a new PR is proper.  
Also, any tip on testing ?